homework/tests/homework.spec.js

```diff
@@ -0,0 +1,172 @@
+      await page.locator('#calculateButton').click();
+  
+      await expect(page.locator('#numberAnswerField')).toHaveValue('23');
+      await page.pause();
+    });
+  });
+});
+
+//Devide by zero error message
+
+data.forEach(version => {
+  test.describe(version + ': Devide by 0', () => {
+    test('Deviding number by 0 getting 0', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('2');
+      await page.locator('#number2Field').type('0');
+      await page.selectOption('#selectOperationDropdown', {label: 'Divide'});
+      await page.locator('#calculateButton').click();
+  
+      await expect(page.locator('#numberAnswerField')).toHaveValue('');
+      await page.pause();
+    });
+  });
+});
+
+//If ts number?
+
+data.forEach(version => {
+  test.describe(version + ': Is it number', () => {
+    test('Check if letter "a" is a number result is not a number', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('a');
+      await page.locator('#number2Field').type('b');
+      await page.selectOption('#selectOperationDropdown', {label: 'Add'});
+      await page.locator('#calculateButton').click();
+  
+      await expect(page.locator('#numberAnswerField')).toBeEmpty();
+      await page.pause();
+    });
+  });
+});
+
+//Not only integer or negative numbers FAAILED PATAISYTI!!!
+
+data.forEach(version => {
+  test.describe(version + ': Does it counts', () => {
+    test(': Multiplying 0.2 by -0.3 results in -0.06', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('0.2');
+      await page.locator('#number2Field').type('-0.3');
+      await page.selectOption('#selectOperationDropdown', {label: 'Multiply'});
+      await page.locator('#calculateButton').click();
+  
+      await expect(page.locator('#numberAnswerField')).toHaveValue('-0.06');
+      await page.pause();
+    });
+  });
+});
+
+//Checkbox "integer"
+
+data.forEach(version => {
+  test.describe(version + ': Checkbox integer', () => {
+    test(': Integer check box should be enabled after clicking', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('1.1');
+      await page.locator('#number2Field').type('2.2');
+      await page.selectOption('#selectOperationDropdown', {label: 'Multiply'});
+      await page.locator('#integerSelect').isEnabled(); //neturi praeiti 4-tame
+      await page.locator('#integerSelect').click(); // patikrinam ar funkcionuoja
```

Comment:
isEnabled() (line above) already checks whether <select> tag element is "functional", so this is not needed.